Source/Libraries/openXDA.Nodes/Types/Analysis/AnalysisNode.cs

```diff
@@ -284,14 +287,17 @@ private void Process(MeterDataSet meterDataSet)
             }
 
             PluginFactory<IDataOperation> dataOperationFactory = new PluginFactory<IDataOperation>();
+            FileGroupProcessingStatus status = FileGroupProcessingStatus.Failed;
+            List<DataOperationFailure> dataOperationFailures = new List<DataOperationFailure>();
+            int index = 0;
 
             foreach (DataOperation dataOperationDefinition in dataOperationDefinitions)
             {
-                if (IsDisposed)
-                    return;
-
                 try
                 {
+                    if (IsDisposed)
```

Comment:
I am pretty sure this will generate multiple failures now. Previously `IsDisposed` caused it to exit the `Process` function completly. Now it will move on to the next `dataOperationDefinition `. Because `IsDisposed` is global for all `dataOperationDefinigitons` we end up with the same error message multiple times. at the very least we should prevent that

Reply:
It will still exit the function completely, but only after logging the exception in the data operation failure. When we catch the exception on line 319, there is a block that checks `IsDisposed `on line 334. If we are, we return out of the function, and never move onto another definition.